### ISSUE TITLE
PDF forms converted using bannertopdf show gibberish text when the form is edited.

### ISSUE THREAD

**Issue by sahilarora535** [OpenPrinting/cups-filters#14]
I created a PDF form by converting a `PDF-BANNER` file. The contents of the banner file are as follows:
```
#PDF-BANNER
Template form_english.pdf
```
The file `form_english.pdf` can be found [here](https://github.com/OpenPrinting/cups-filters/blob/master/data/form_english.pdf).

The banner file is converted to PDF using the following command:
```
cupsfilter banner_file > OUT.pdf
```

In the original file, if a field in a form is edited, it gets correctly edited and is reflected in the file. However, in the converted PDF file `OUT.pdf`, the form fields in the file get filled with gibberish as soon as the field is clicked on, or if tried to be edited. See here sample image. The field 'Task name' was edited and this is the final result:

![image](https://user-images.githubusercontent.com/12633577/34456564-900b3ad4-edbf-11e7-9ee3-b829ad8e26d5.png)

Here is the log for the conversion:

```
DEBUG: argv[0]="cupsfilter"
DEBUG: argv[1]="1"
DEBUG: argv[2]="sahil"
DEBUG: argv[3]="default"
DEBUG: argv[4]="1"
DEBUG: argv[5]=""
DEBUG: argv[6]="default"
DEBUG: envp[0]="<CFProcessPath>"
DEBUG: envp[1]="CONTENT_TYPE=application/vnd.cups-pdf-banner"
DEBUG: envp[2]="CUPS_DATADIR=/usr/share/cups"
DEBUG: envp[3]="CUPS_FONTPATH=/usr/share/cups/fonts"
DEBUG: envp[4]="CUPS_SERVERBIN=/usr/lib/cups"
DEBUG: envp[5]="CUPS_SERVERROOT=/etc/cups"
DEBUG: envp[6]="LANG=en_US.UTF8"
DEBUG: envp[7]="PATH=/usr/lib/cups/filter:/usr/bin:/usr/sbin:/bin:/usr/bin"
DEBUG: envp[8]="PPD=/usr/share/cups/model/laserjet.ppd"
DEBUG: envp[9]="PRINTER_INFO=cupsfilter"
DEBUG: envp[10]="PRINTER_LOCATION=Unknown"
DEBUG: envp[11]="PRINTER=cupsfilter"
DEBUG: envp[12]="RIP_MAX_CACHE=128m"
DEBUG: envp[13]="USER=sahil"
DEBUG: envp[14]="CHARSET=utf-8"
DEBUG: envp[15]="FINAL_CONTENT_TYPE=application/pdf"
INFO: bannertopdf (PID 11417) started.
DEBUG: Could not open PPD file '/usr/share/cups/model/laserjet.ppd'
Resource dict was changed.
Fill widget name job-id with value 1.
Fill widget name job-title with value default.
Fill widget name user with value sahil.
Fill widget name printer-name with value cupsfilter.
Fill widget name printer-info with value cupsfilter.
Fill widget name time-at-creation with value unknown.
Fill widget name time-at-processing with value unknown.
Lack information for widget: job-originating-host-name.
Fill widget name job-originating-host-name with value N/A.
Lack information for widget: banner-font.
Fill widget name banner-font with value N/A.
Lack information for widget: banner-font-size.
Fill widget name banner-font-size with value N/A.
Lack information for widget: job-uuid.
Fill widget name job-uuid with value N/A.
Lack information for widget: driver.
Fill widget name driver with value N/A.
Lack information for widget: driver-version.
Fill widget name driver-version with value N/A.
Lack information for widget: make-and-model.
Fill widget name make-and-model with value N/A.
Lack information for widget: security-context-user.
Fill widget name security-context-user with value N/A.
Lack information for widget: security-context-range-cur.
Fill widget name security-context-range-cur with value N/A.
Lack information for widget: security-context-type.
Fill widget name security-context-type with value N/A.
INFO: bannertopdf (PID 11417) exited with no errors.
```

**Comment by tillkamppeter** [comment]
Sahil, it was not me who wrote bannertopdf so I do not know how it exactly works internally. I assume that it worked correctly when it was written several years ago, and kept working for a long time as there were no other complaints before yours. For me it looks like that changes in the unstable interfaces of Poppler made it fail.
As you are already working on replacing the Poppler-based bannertopdf by a QPDF-based one, it's not worth the effort any more to fix the current bannertopdf.
So simply make the new bannertopdf with these things working.

**Comment by tillkamppeter** [comment]
Should be solved by the solution of issue #8.
